dask/dataframe/core.py

```diff
@@ -2600,6 +2604,32 @@ def clip_lower(self, threshold):
     def clip_upper(self, threshold):
         return self.map_partitions(M.clip_upper, threshold=threshold)
 
+    @derived_from(pd.DataFrame)
+    def squeeze(self, axis=None):
+        if axis in [None, 1]:
+            squeezed_series = self[self.columns[0]]
+
+            if len(self.columns) == 1:
+                if len(self.index) == 1:
+                    # dask dataframe with only one element should return scalar
+                    dask_arr = squeezed_series.values
+                    # set chunks so we can index for first and only element
+                    dask_arr._chunks = ((1, ),)
```

Comment:
It's this line that is non-lazy:

    if len(self.index) == 1:

Dataframes don't know how long they are without computing themselves.  I recommend that we don't support the pandas scalar case.